foundations/javascript_basics/object_basics.md

```diff
@@ -20,6 +20,94 @@ Objects are a *very* important part of the JavaScript language, and while for th
 1. This JavaScript.info [article on objects](https://javascript.info/object) is the best place to get started.
 1. The [MDN tutorial on objects](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Basics) isn't bad either, so check it out if you need another take on the subject.
 
+### Differences between objects and primitives
+
+Earlier in the curriculum you've learned about [primitive data types](https://www.theodinproject.com/lessons/foundations-data-types-and-conditionals). Now that you've seen the object data type, which includes but is not limited to, objects ({key: value}), arrays, and functions. The main difference between the two is that primitives can contain only a single thing (string, number, etc). Objects data types are used to store collection of data and more complex entities.
+
+Besides the formal differences, there are also some technical differences, which affect how we use each data type.
+When you define a primitive variable, it will contain a copy of the information provided to it:
+
+```javascript
+let data = 42;
+// dataCopy will store a copy of what data contains, so a copy of 42
+let dataCopy = data;
+
+// which means that making changes to dataCopy won't affect data
+dataCopy = 43;
+
+console.log(data); // 42
+console.log(dataCopy); // 43
+```
+
+On the other hand, when you define an object variable, it will contain a *reference* to the object provided to it:
+
+```javascript
+// obj contains a reference to the object we defined on the right side
+const obj = { data: 42 };
+// objCopy will contain a reference to the object referenced by obj
+const objCopy = obj;
+
+// making changes to objCopy will make changes to the object that it refers to
+objCopy.data = 43;
+
+console.log(obj); // { data: 43 }
+console.log(objCopy); // { data: 43 }
+```
+
+This behavior isn't new to you. In your last project, you made changes to the cells in the Etch-A-Sketch grid by using references.
+Let's take this code snippet as an example:
+
+```javascript
+const element = document.querySelector("#container");
+element.style.backgroundColor = "red";
+```
+
+We're mutating the variable we declared (`element`), yet the changes affect the corresponding node in the DOM.
+Why does it happen?
+That's because the node we have in our code is a **reference** to the same node that our DOM uses. If that wasn't a reference, but a copy like primitive data types behave, our changes would have **no** effect! Because the changes would be made to the local copy we have.
```

Comment:
Single line breaks don't do anything different, so we keep these sentences on the same line (unless you want to separate into paragraphs, in which case separate with a single blank line).
```suggestion
We're mutating the variable we declared (`element`), yet the changes affect the corresponding node in the DOM. Why does it happen? That's because the node we have in our code is a **reference** to the same node that our DOM uses. If that wasn't a reference, but a copy like primitive data types behave, our changes would have **no** effect! Because the changes would be made to the local copy we have.
```